### ISSUE TITLE
Design Patterns used by AB3

### ISSUE THREAD

**Issue by michaelyql** [nus-cs2103-AY2425S1/forum#813]
Would it be right to say that AB3 uses the MVC, Observer, Facade and Singleton design patterns? 

MVC, Observer are straightforward and obvious.

For facade, I believe that `Storage` is an example of a Facade class. For example, `LogicManager` calls `storage.saveAddressBook(...)` after the execution of each command. It does not know about the details of how storage saves to disk. `Storage` is providing an interface here for `LogicManager` to use. 

For Singleton, am I right to say that the logger used in `MainApp` is an example of a singleton? Since it is declared `private static final` and is initialized using a factory method from `LogsCenter` which controls the instantiation of `Logger` objects.

**Comment by rahula1008** [comment]
Hi @michaelyql 

So if I'm not wrong, the logger you're referring to is a variable, and the Singleton design pattern typically refers to a class which is set up in such a way that there can only be one instance, i.e the class will have a private constructor, a private static theOne variable, and a getInstance method that gets that Singleton instance for you.

By any chance, were you trying to ask whether the LogsCenter class is a singleton? If so, I think actually, the LogsCenter class is not a Singleton, as firstly, it's `init()` method(which acts as its constructor) is a public method. Secondly, there is no instance in the LogsCenter class which holds the Singleton, i.e a `theOne` variable.  

Hope that helps!

**Comment by michaelyql** [comment]
@rahula1008 
No, I was referring the to `logger` variable in `MainApp.java`. 

### `MainApp`
<img width="796" alt="Screenshot 2024-11-19 at 12 17 00 PM" src="https://github.com/user-attachments/assets/7c283c99-91a4-46c8-8d67-2c6a30588754">

### Methods in `LogsCenter`
<img width="900" alt="Screenshot 2024-11-19 at 12 17 21 PM" src="https://github.com/user-attachments/assets/800b8d7a-d9cf-4162-81ee-0c444347a211">

If you trace the method call to the core Java implementation for `Logger`, you will see that a `new Logger(...)` instance is created.
